src/MetaMorpho.sol

```diff
@@ -738,17 +738,13 @@ contract MetaMorpho is ERC4626, ERC20Permit, Ownable2Step, Multicall, IMetaMorph
 
         if (supplyCap > 0) {
             if (!marketConfig.enabled) {
-                supplyQueue.push(id);
                 withdrawQueue.push(id);
 
-                if (
-                    supplyQueue.length > ConstantsLib.MAX_QUEUE_LENGTH
-                        || withdrawQueue.length > ConstantsLib.MAX_QUEUE_LENGTH
-                ) {
-                    revert ErrorsLib.MaxQueueLengthExceeded();
-                }
+                if (withdrawQueue.length > ConstantsLib.MAX_QUEUE_LENGTH) revert ErrorsLib.MaxQueueLengthExceeded();
 
                 marketConfig.enabled = true;
+
+                emit EventsLib.SetWithdrawQueue(msg.sender, withdrawQueue);
```

Comment:
I am not sure it is a real issue
But if it is, we could also create a dedicated issue `AppendWithdrawQueue` to decrease the gas cost (because this line alone is responsible for 1 event + N cold SLOADs = 1k + N * 2.1k)

Reply:
@julien-devatom

Reply:
Seen w @julien-devatom , it seems fine to have another event `AppendWithdrawQueue` to limit gas cost if we want to do so

Reply:
whatever fits me tbh